Submitted By:            DJ Lucas <dj_AT_linuxfromscratch_DOT_org>
Date:                    2017-08-22
Initial Package Version: 60.0.3112.101
Upstream Status:         Committed
Origin:                  https://hg.mozilla.org/integration/autoland/rev/d3970338a22b
Description:             Allows building with GlibC-2.6.


diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc chromium-60.0.3112.101/breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc	2017-08-21 23:26:02.474938230 -0500
@@ -40,11 +40,11 @@ namespace google_breakpad {
 
 #if defined(__i386__)
 
-uintptr_t UContextReader::GetStackPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetStackPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.gregs[REG_ESP];
 }
 
-uintptr_t UContextReader::GetInstructionPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetInstructionPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.gregs[REG_EIP];
 }
 
@@ -88,15 +88,15 @@ void UContextReader::FillCPUContext(RawC
 
 #elif defined(__x86_64)
 
-uintptr_t UContextReader::GetStackPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetStackPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.gregs[REG_RSP];
 }
 
-uintptr_t UContextReader::GetInstructionPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetInstructionPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.gregs[REG_RIP];
 }
 
-void UContextReader::FillCPUContext(RawContextCPU *out, const ucontext *uc,
+void UContextReader::FillCPUContext(RawContextCPU *out, const ucontext_t *uc,
                                     const struct _libc_fpstate* fpregs) {
   const greg_t* regs = uc->uc_mcontext.gregs;
 
@@ -145,11 +145,11 @@ void UContextReader::FillCPUContext(RawC
 
 #elif defined(__ARM_EABI__)
 
-uintptr_t UContextReader::GetStackPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetStackPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.arm_sp;
 }
 
-uintptr_t UContextReader::GetInstructionPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetInstructionPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.arm_pc;
 }
 
@@ -184,11 +184,11 @@ void UContextReader::FillCPUContext(RawC
 
 #elif defined(__aarch64__)
 
-uintptr_t UContextReader::GetStackPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetStackPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.sp;
 }
 
-uintptr_t UContextReader::GetInstructionPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetInstructionPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.pc;
 }
 
@@ -210,11 +210,11 @@ void UContextReader::FillCPUContext(RawC
 
 #elif defined(__mips__)
 
-uintptr_t UContextReader::GetStackPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetStackPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.gregs[MD_CONTEXT_MIPS_REG_SP];
 }
 
-uintptr_t UContextReader::GetInstructionPointer(const struct ucontext* uc) {
+uintptr_t UContextReader::GetInstructionPointer(const ucontext_t* uc) {
   return uc->uc_mcontext.pc;
 }
 
diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h chromium-60.0.3112.101/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/dump_writer_common/ucontext_reader.h	2017-08-21 23:26:02.474938230 -0500
@@ -41,21 +41,21 @@ namespace google_breakpad {
 
 // Wraps platform-dependent implementations of accessors to ucontext structs.
 struct UContextReader {
-  static uintptr_t GetStackPointer(const struct ucontext* uc);
+  static uintptr_t GetStackPointer(const ucontext_t* uc);
 
-  static uintptr_t GetInstructionPointer(const struct ucontext* uc);
+  static uintptr_t GetInstructionPointer(const ucontext_t* uc);
 
   // Juggle a arch-specific ucontext into a minidump format
   //   out: the minidump structure
   //   info: the collection of register structures.
 #if defined(__i386__) || defined(__x86_64)
-  static void FillCPUContext(RawContextCPU *out, const ucontext *uc,
+  static void FillCPUContext(RawContextCPU *out, const ucontext_t *uc,
                              const struct _libc_fpstate* fp);
 #elif defined(__aarch64__)
-  static void FillCPUContext(RawContextCPU *out, const ucontext *uc,
+  static void FillCPUContext(RawContextCPU *out, const ucontext_t *uc,
                              const struct fpsimd_context* fpregs);
 #else
-  static void FillCPUContext(RawContextCPU *out, const ucontext *uc);
+  static void FillCPUContext(RawContextCPU *out, const ucontext_t *uc);
 #endif
 };
 
diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/handler/exception_handler.cc chromium-60.0.3112.101/breakpad/src/client/linux/handler/exception_handler.cc
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/handler/exception_handler.cc	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/handler/exception_handler.cc	2017-08-21 23:26:02.474938230 -0500
@@ -444,9 +444,9 @@ bool ExceptionHandler::HandleSignal(int
   // Fill in all the holes in the struct to make Valgrind happy.
   memset(&g_crash_context_, 0, sizeof(g_crash_context_));
   memcpy(&g_crash_context_.siginfo, info, sizeof(siginfo_t));
-  memcpy(&g_crash_context_.context, uc, sizeof(struct ucontext));
+  memcpy(&g_crash_context_.context, uc, sizeof(ucontext_t));
 #if defined(__aarch64__)
-  struct ucontext* uc_ptr = (struct ucontext*)uc;
+  ucontext_t* uc_ptr = (ucontext_t*)uc;
   struct fpsimd_context* fp_ptr =
       (struct fpsimd_context*)&uc_ptr->uc_mcontext.__reserved;
   if (fp_ptr->head.magic == FPSIMD_MAGIC) {
@@ -455,9 +455,9 @@ bool ExceptionHandler::HandleSignal(int
   }
 #elif !defined(__ARM_EABI__) && !defined(__mips__)
   // FP state is not part of user ABI on ARM Linux.
-  // In case of MIPS Linux FP state is already part of struct ucontext
+  // In case of MIPS Linux FP state is already part of ucontext_t
   // and 'float_state' is not a member of CrashContext.
-  struct ucontext* uc_ptr = (struct ucontext*)uc;
+  ucontext_t* uc_ptr = (ucontext_t*)uc;
   if (uc_ptr->uc_mcontext.fpregs) {
     memcpy(&g_crash_context_.float_state, uc_ptr->uc_mcontext.fpregs,
            sizeof(g_crash_context_.float_state));
@@ -481,7 +481,7 @@ bool ExceptionHandler::SimulateSignalDel
   // ExceptionHandler::HandleSignal().
   siginfo.si_code = SI_USER;
   siginfo.si_pid = getpid();
-  struct ucontext context;
+  ucontext_t context;
   getcontext(&context);
   return HandleSignal(sig, &siginfo, &context);
 }
diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/handler/exception_handler.h chromium-60.0.3112.101/breakpad/src/client/linux/handler/exception_handler.h
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/handler/exception_handler.h	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/handler/exception_handler.h	2017-08-21 23:26:02.474938230 -0500
@@ -191,7 +191,7 @@ class ExceptionHandler {
   struct CrashContext {
     siginfo_t siginfo;
     pid_t tid;  // the crashing thread.
-    struct ucontext context;
+    ucontext_t context;
 #if !defined(__ARM_EABI__) && !defined(__mips__)
     // #ifdef this out because FP state is not part of user ABI for Linux ARM.
     // In case of MIPS Linux FP state is already part of struct
diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/microdump_writer/microdump_writer.cc chromium-60.0.3112.101/breakpad/src/client/linux/microdump_writer/microdump_writer.cc
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/microdump_writer/microdump_writer.cc	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/microdump_writer/microdump_writer.cc	2017-08-21 23:26:58.394905296 -0500
@@ -593,7 +593,7 @@ class MicrodumpWriter {
 
   void* Alloc(unsigned bytes) { return dumper_->allocator()->Alloc(bytes); }
 
-  const struct ucontext* const ucontext_;
+  const ucontext_t* const ucontext_;
 #if !defined(__ARM_EABI__) && !defined(__mips__)
   const google_breakpad::fpstate_t* const float_state_;
 #endif
diff -Naurp chromium-60.0.3112.101-orig/breakpad/src/client/linux/minidump_writer/minidump_writer.cc chromium-60.0.3112.101/breakpad/src/client/linux/minidump_writer/minidump_writer.cc
--- chromium-60.0.3112.101-orig/breakpad/src/client/linux/minidump_writer/minidump_writer.cc	2017-08-14 14:06:36.000000000 -0500
+++ chromium-60.0.3112.101/breakpad/src/client/linux/minidump_writer/minidump_writer.cc	2017-08-21 23:26:02.474938230 -0500
@@ -1323,7 +1323,7 @@ class MinidumpWriter {
   const int fd_;  // File descriptor where the minidum should be written.
   const char* path_;  // Path to the file where the minidum should be written.
 
-  const struct ucontext* const ucontext_;  // also from the signal handler
+  const ucontext_t* const ucontext_;  // also from the signal handler
 #if !defined(__ARM_EABI__) && !defined(__mips__)
   const google_breakpad::fpstate_t* const float_state_;  // ditto
 #endif
